### PR TITLE
Update eslint-plugin-react-hooks to latest major release

### PR DESCRIPTION
**Pull request by bjankord**
### Summary
Update eslint-plugin-react-hooks to latest major release.

Related: https://github.com/facebook/react/pull/16528/files

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
     "eslint-plugin-import": "^2.14.0",
     "eslint-plugin-jsx-a11y": "^6.1.1",
     "eslint-plugin-react": "^7.12.2",
-    "eslint-plugin-react-hooks": "^1.0.0"
+    "eslint-plugin-react-hooks": "^2.0.1"
```
Comment (StephenEsser) [inline package.json]: Are we major version bumping this package again?

Comment (bjankord) [reply package.json]: [My understanding was this version adds a lint rule around usage of hooks at top level which crashes the app](https://github.com/facebook/react/pull/16455). 

So before if you used hooks at top level; the app will crash. With pulling in this version now, if you use hooks at top level, you'll get a lint warning + app still crashes.

Based on that I'm leaning towards treating this as a minor release, however the react team [considered this a breaking change](https://github.com/facebook/react/pull/16455#issuecomment-522707242).

Comment (emilyrohrbough) [reply package.json]: my vote is ship it as a patch release lol

**Comment by dev-cprice** [comment]
Curious why we don't use the full recommended config, namely we left off the rule:

`"react-hooks/exhaustive-deps": "warn"`

This rule is pretty important and suggests fixes for common pitfalls involving stale hook dependencies. Commentary [here](https://github.com/facebook/react/issues/14920)

**Comment by bjankord** [comment]
@dev-cprice Added config for exhaustive-deps in this commit: https://github.com/cerner/eslint-config-terra/pull/37/commits/2173277485513fae76d548cc82454835ba4a11c5

**Pull request merged**